mesh/mesh.go

```diff
@@ -308,15 +306,12 @@ func (msh *Mesh) processValidityUpdates(ctx context.Context, logger log.Log, upd
 	// upon restarts, tortoise will use persisted validity, and there is no way for mesh
 	// to determine whether a state revert should be done.
 	if err := msh.cdb.WithTx(ctx, func(dbtx *sql.Tx) error {
-		for lid, updates := range msh.validityUpdates {
-			for bid, valid := range updates {
-				logger.With().Info("save block contextual validity",
-					bid, lid, log.Bool("validity", valid))
-				if valid {
-					if err := blocks.SetValid(dbtx, bid); err != nil {
-						return err
-					}
-				} else if err := blocks.SetInvalid(dbtx, bid); err != nil {
+		for i := range results {
+			for _, block := range results[i].Blocks {
+				if !block.Data {
+					continue
```

Comment:
i think we should return error here instead of continuing. 

from the code logic, since it errors out when there are missing blocks, this should not happen. 
this is the only place in code that block validity is updated. if indeed there are missing blocks at this point, the block validity will not be set ever if the code continue.

Reply:
we don't ask peers for any block that crossed negative threshold. it is probably one of the main reasons for doing this change